### DISCUSSION TITLE
Detecting deleted and blocked users

### DISCUSSION THREAD

**Discussion by kerimovok** [pocketbase/pocketbase#3298] [Q&A] [answered]
Hello, I want to use an axios interceptor to check the response to the status code 401 (ex. I deleted user from Admin UI, but in user still can acces frontend. Yes without any data (404 codes in console), but it must not happen, user shoud be logged out), so I am logging out user (pb.authStore.clear())
I did it in my last projects, smth like this:
```js
axios.interceptors.response.use(undefined, function (error) {
    if (error.response && error.response.status === 401) {
        store.dispatch("clearToken").catch((error) => {
                console.error("Failed to clear token:", error);
        });
        router.push("/login");
    }
    return Promise.reject(error);
});
```

But it didn't work, because as I can see pocketbase only returns 404 response, I read lot of discussins there and got some similar code and tried it:
```js
axios.interceptors.response.use(undefined, async function (error) {
  try {
    pb.authStore.isValid && (await pb.collection('users').authRefresh())
  } catch (e) {
    pb.authStore.clear()
    await router.push('/login')
  }
  return Promise.reject(error)
})
```

But it didn't helped either, is there any way to make it worked?

And will there be the field for blocking the user be added in the future to users collection, because right now I must write custom middleware or use filters which will return 404?

**Comment by ganigeorgiev** [comment]
I'm confused. Where and what triggers the above interceptor?
Why do you use axios in the first place (note that the SDK uses `fetch`)?

`authRefresh()` will return 401 in case it fails, eg. on invalid/missing token or user.
You can either track the user realtime changes with `collection('users').subscribe('user_id', (e) => { ... })` OR if you want to check whether the token is valid or not, you can call `authRefresh()` when your request fails. If `authRefresh()` also fails then either the token is invalid or the user doesn't exist, aka. in both cases the user can be redirected to the login page.

**Reply by kerimovok** [reply]
Axios interceptors are functions that are called before a request is sent and after a response is received. So, any status codes that falls outside the range of 2xx cause this function to trigger. If I understood you right, so my second code shoud work right? But, as I see it isn't(
I dont actually want to subscribe only because of this. I will try to use IF statement and authRefresh() maybe it will help. Thank you!
I would also appreciate it if you answer the last question!

**Reply by ganigeorgiev** [reply]
But the JS SDK doesn't use axios, aka. the axios interceptors will not get triggered when sending request with `fetch` or the SDK (_unless axios overwrite the global `fetch` function, but I doubt it_).

To confirm it, you can try to add a `console.log` and check if it is called.

**Reply by kerimovok** [reply]
Ah, now I understand you, I'm sorry, I just always used axios and didn't even think about it, what can I do in this case so that I can solve my problem? Try catch authRefresh when I am fetching one of the collections?

**Reply by kerimovok** [reply, marked as answer]
Something like this?
(I am using Pinia and VueJS)

PS: I realized that this will not work, because authStore is not changing if user is deleted from Admin UI, but I will try to use on other collection

PS2: That actually worked, I just calling fetchUser in my MainLayout (onMounted)

```js
  const user: Ref<UsersResponse | undefined> = ref()
  const userModel: Ref<AuthModel> = ref(pb.authStore.model)

  pb.authStore.onChange(async (): Promise<void> => {
    userModel.value = pb.authStore.model
    await fetchUser()
  })
  
  async function fetchUser(): Promise<void> {
    user.value = await pb
      .collection(Collections.Users)
      .getOne<UsersResponse>(pb.authStore.model!.id)
      .catch(async (): Promise<void> => {
        try {
          pb.authStore.isValid && (await pb.collection('users').authRefresh())
        } catch (e) {
          pb.authStore.clear()
          await router.push('/login')
        }
      })
  }
```

**Reply by ganigeorgiev** [reply]
Yes, it depends on your use case but this could be one possible solution.

Alternatively, since the JS SDK doesn't have a global error interceptor at the moment, another slightly hacky workaround could be to try overwritting the `send` method and catch/filter all errors from there, something like (_I haven't tested it_):

```js
import PocketBase from 'pocketbase';

const originalSend = PocketBase.prototype.send;
PocketBase.prototype.send = async function (path, options) {
    return originalSend.call(this, path, options).catch(async (err) => {
        // do something with the error, eg. check the status code
        if (err.status == 404) {
            // refresh the auth state
            try {
                this.authStore.isValid && await this.collection('users').authRefresh();
            } catch (err) {
                this.authStore.clear();
            }
        } else {
            // rethrow
            throw err;
        }
    });
}

const pb = new PocketBase("http://127.0.0.1:8090")
```

_edit: updated the originalSend to be called with the local this._
